sessionCheck.js

```diff
@@ -6,77 +6,160 @@
+     * @param {string} [config.responseType=id_token] - Response type to use to check the session. Supported options are id_token or none.
      * @param {string} config.clientId - The id of this RP client within the OP
      * @param {string} config.opUrl - Full URL to the OP Authorization Endpoint
      * @param {function} config.invalidSessionHandler - function to be called once any problem with the session is detected
-     * @param {string} config.redirectUri [sessionCheck.html] - The redirect uri registered in the OP for session-checking purposes
-     * @param {number} config.cooldownPeriod [5] - Minimum time (in seconds) between requests to the opUrl
-     * @param {string} config.scope [openid] - Session check scope; can be space-separated list
+     * @param {function} [config.initialSessionSuccessHandler] - optional function to be called after the first successful session check request.
+     * @param {function} [config.sessionClaimsHandler] - optional function to be called after every session check request. Only invoked when using responseType=id_token. Includes the claims and the count of session check requests that have been made so far.
+     * @param {string} [config.redirectUri=sessionCheck.html] - The redirect uri registered in the OP for session-checking purposes
+     * @param {number} [config.cooldownPeriod=5] - Minimum time (in seconds) between requests to the opUrl
+     * @param {string} [config.scope=openid] - Session check scope; can be space-separated list
      */
     module.exports = function (config) {
         var calculatedUriLink;
 
-        if (!config.redirectUri) {
-            // trick used to make the DOM resolve a complete URI based on the relative URI
-            calculatedUriLink = document.createElement("a");
-            calculatedUriLink.href = "sessionCheck.html";
+        this.request_check_count = 0;
+        this.cooldownPeriod = config.cooldownPeriod || 5;
+        this.subject = config.subject;
 
-            this.redirectUri = calculatedUriLink.href;
-        } else {
-            this.redirectUri = config.redirectUri;
-        }
+        if (config.ssoToken) {
+            this.ssoTokenName = config.ssoTokenName || "iPlanetDirectoryPro";
+            this.ssoToken = config.ssoToken;
+            this.amUrl = config.amUrl + "/sessions?_action=validate";
+            this.validationHandler = (function (responseText) {
+                try {
+                    var validationResponse = JSON.parse(responseText);
+                } catch (e) {
+                    config.invalidSessionHandler(e.message, this.request_check_count);
+                    return;
+                }
+                if (!validationResponse.valid) {
+                    config.invalidSessionHandler("invalid_session", this.request_check_count);
+                    return;
+                }
+                if (this.subject && this.subject !== validationResponse.uid) {
```

Comment:
I don't think there is any real need to over-complicate this. The id_token really only provided to this library so we can perform the response_type=none option. Since we aren't using that option, simply accepting the sub (which can be extracted from an id_token by the app developer) is enough.